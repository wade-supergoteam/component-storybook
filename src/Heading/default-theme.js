export const headingsMarginBottom = ".5rem";
export const headingsFontFamily = "inherit";
export const headingsFontWeight = "500";
export const headingsLineHeight = "1.2";
export const headingsColor = "inherit";

export const fontSizeBase = "1rem";
export const h1FontSize = "2.5rem";
export const h2FontSize = "2rem";
export const h3FontSize = "1.75rem";
export const h4FontSize = "1.5rem";
export const h5FontSize = "1.25rem";
export const h6FontSize = "1rem";
