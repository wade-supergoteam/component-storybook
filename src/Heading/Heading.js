import { createElement } from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import omit from "lodash.omit";

import themeProp from "../utils/theme";

import * as defaultTheme from "./default-theme";

const { headingsMarginBottom, headingsFontFamily, headingsFontWeight, headingsLineHeight, headingsColor } = defaultTheme;

const Element = ({ as, children, ...props }) => createElement(as, omit(props, ["as", "size"]), children);

const Heading = styled(Element)`
  margin-bottom: ${themeProp("headingsMarginBottom", headingsMarginBottom)};
  font-family: ${themeProp("headingsFontFamily", headingsFontFamily)};

  font-weight: ${themeProp("headingsFontWeight", headingsFontWeight)};
  line-height: ${themeProp("headingsLineHeight", headingsLineHeight)};
  color: ${themeProp("headingsColor", headingsColor)};

  ${props => {
    const size = props.size || props.as;

    return css`
      font-size: ${themeProp(`${size}FontSize`, defaultTheme[`${size}FontSize`])};
    `;
  }};
`;

Heading.defaultProps = {
  as: "h1",
};

Heading.proptypes = {
  // as: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'span', 'p']),
  size: PropTypes.oneOf(["h1", "h2", "h3", "h4", "h5", "h6"]),
};

export default Heading;
