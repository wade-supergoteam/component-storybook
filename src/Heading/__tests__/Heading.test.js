import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import Heading from "..";

test("Create H1 Default", () => {
  const wrapper = renderer.create(<Heading as="h1" />);
  expect(wrapper).toMatchSnapshot();
});

test("Create H1 With sizes, size H6", () => {
  const wrapper = renderer.create(<Heading as="h1" size="h6" />);
  expect(wrapper).toMatchSnapshot();
});

test("Create h2 with sizes, size h5", () => {
  const wrapper = renderer.create(<Heading as="h2" size="h5" />);
  expect(wrapper).toMatchSnapshot();
});

test("Create h3 with sizes, size H4", () => {
  const wrapper = renderer.create(<Heading as="h3" size="h4" />);
  expect(wrapper).toMatchSnapshot();
});

test("Create h4 with sizes, size H3", () => {
  const wrapper = renderer.create(<Heading as="h4" size="h3" />);
  expect(wrapper).toMatchSnapshot();
});

test("Create h5 with sizes, size H5", () => {
  const wrapper = renderer.create(<Heading as="h5" size="h2" />);
  expect(wrapper).toMatchSnapshot();
});

test("Create h6 with sizes, size H1", () => {
  const wrapper = renderer.create(<Heading as="h6" size="h6" />);
  expect(wrapper).toMatchSnapshot();
});

test("Create h6 with sizes, size H1", () => {
  const wrapper = renderer.create(
    <Heading as="h6" size="h6">
      With child element
    </Heading>,
  );
  expect(wrapper).toMatchSnapshot();
});
