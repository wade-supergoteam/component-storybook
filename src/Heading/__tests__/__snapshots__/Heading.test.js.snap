// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create H1 Default 1`] = `
<h1
  className="sc-bdVaJa fCrnLG"
/>
`;

exports[`Create H1 With sizes, size H6 1`] = `
<h1
  className="sc-bdVaJa dAipnw"
  size="h6"
/>
`;

exports[`Create H1 with sizes, size H1 1`] = `
<h6
  className="sc-bdVaJa fCrnLG"
  size="h1"
/>
`;
